### ISSUE TITLE
Simulation stops without an event at the end

### ISSUE THREAD

**Issue by andreabedini** [dsorokin/aivika#5]
Hi @dsorokin, thank you for your beautiful package.

I am puzzled by the following behaviour. Without the `runEventInStopTime` then simulation stops after showing "t = 0.0 doing xyz" once. With the `runEventInStopTime` the simulation keeps printing "doing xyz" at every second, which is what I would have expected. Would you be able to explain the origin of this behaviour? Thanks in advance.

```
import Simulation.Aivika

specs :: Specs
specs =
    Specs
        { spcStartTime = 0.0
        , spcStopTime = 10.0
        , spcDT = 0.1
        , spcMethod = RungeKutta4
        , spcGeneratorType = SimpleGenerator
        }

test :: IO ()
test = runSimulation model specs

model :: Simulation ()
model = do
  let myProcess = traceProcess "doing xyz" $ holdProcess 1 >> myProcess

  runProcessInStartTime myProcess

  -- without the following line the process doesn't resume after `holdProcess`
  -- runEventInStopTime $ return ()
```

**Comment by andreabedini** [comment]
I dug into the source code and I think I am getting it now. Basically there's nothing that kicks off the processing of the event queue. If it was exported, one could do `processEvents CurrentEvents` to kick off the events. Still the event queue is a property of the simulation run, I am a bit disappointed `Simulation` is not capable of running the events itself.

Am I on the right track?

**Comment by dsorokin** [comment]
Yes, the Simulation computaion has no any deal with the event queue except for one thing: this computation creates an event queue instance, but it doesn't use it. The computation is not bound with any specific modeling time point, but the Simulation computation is related to the entire simulation run at once. Therefore, it is necessary to call some event handler at fhe final time point of simulation. Usually, there is no any problem with it if you use so called simulation experiments [1], which implies Monte Carlo method and charting. Such experiments do indeed call event handlers at the stop time themselves, which effects on the event queue.  This all is a part of design. The Simulation computation must not call implicitly the event queue itself. This is an exclusive prerogative of the Even computation, which is designed specifically for working with the event queue.

[1]  packages aivika-experiment and aivika-experiment-chart as described in my book about Aivika

**Comment by andreabedini** [comment]
Thanks @dsorokin
